### PR TITLE
Fixes #35152 - Add clone media option

### PR DESCRIPTION
**Pull request by Dyrkon**
Clone media option added to Installation Media section.

### CODE REVIEW

**Comment by stejskalleos** [comment]
ok to test

**Comment by ofedoren** [comment]
Test failure seems to be related :(

**Comment by stejskalleos** [comment]
[test katello]

File: test/controllers/media_controller_test.rb [hunk 1]
```diff
@@ -58,4 +58,18 @@ def setup_user
     get :index, session: set_session_user.merge(:user => users(:one).id)
     assert_response :success
   end
+
+  test 'user should be able to clone media' do
+    medium = media(:solaris10)
+    setup_user
+    get :clone, params: { :id => medium.id}, session: set_session_user
```
Comment (stejskalleos) [inline test/controllers/media_controller_test.rb]: Nit: `{ id: medium.id }`

File: test/controllers/media_controller_test.rb [hunk 2]
```diff
@@ -58,4 +58,18 @@ def setup_user
...
+    assert_response :success
+  end
+
+  test 'user without correct permissions shouldnt clone media' do
+    medium = media(:solaris10)
+    setup_user
+    get :clone, params: { :id => medium.id}, session: set_session_user.merge(:user => users(:one).id)
```
Comment (stejskalleos) [inline test/controllers/media_controller_test.rb]: Nit: `{ id: medium.id }`

File: test/controllers/media_controller_test.rb [hunk 3]
```diff
@@ -58,4 +58,18 @@ def setup_user
...
   end
+
+  test 'user should be able to clone media' do
+    medium = media(:solaris10)
+    setup_user
+    get :clone, params: { :id => medium.id}, session: set_session_user
+    assert_response :success
+  end
+
+  test 'user without correct permissions shouldnt clone media' do
```
Comment (stejskalleos) [inline test/controllers/media_controller_test.rb]: Wording: `"user without create_media permission can't clone media"`

File: app/controllers/media_controller.rb [hunk 4]
```diff
@@ -39,4 +39,9 @@ def destroy
       process_error
     end
   end
+
+  def clone
+    @medium = Medium.find(params[:id]).dup
```
Comment (stejskalleos) [inline app/controllers/media_controller.rb]: You can add `:clone` to `before_action` and then just do `@medium.dup`

**Review by stejskalleos — CHANGES_REQUESTED** [review]

File: app/controllers/media_controller.rb (hunk 4 above)
Comment (Dyrkon) [reply app/controllers/media_controller.rb]: Doing so results in a following error: ERF42-5531 [Foreman::Exception]: unknown permission for media#clone
Should I change the permissions (in that case I'll need a hint where to find them) or leave it be?

**Comment by Dyrkon** [comment]
@stejskalleos After implementing your suggestion and tweaking the permissions, the find_resource before_action does not seem to work correctly.

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -167,7 +167,7 @@ def controller_permission
 
   def action_permission
     case params[:action]
-      when 'new', 'create'
+      when 'new', 'create', 'clone'
```
Comment (stejskalleos) [inline app/controllers/application_controller.rb]: This should be in the media_controller, not here.
Same as https://github.com/theforeman/foreman/pull/9397/files#r963420270

File: app/controllers/media_controller.rb [hunk 6]
```diff
@@ -39,4 +39,9 @@ def destroy
       process_error
     end
   end
+
+  def clone
+    @medium = @media.dup
```
Comment (stejskalleos) [inline app/controllers/media_controller.rb]: Typo: `@medium = @medium.dup`

**Review by stejskalleos — CHANGES_REQUESTED** [review]

**Comment by stejskalleos** [comment]
[test katello]

**Review by stejskalleos — APPROVED** [review]
:green_apple: **LGTM**

Thanks @Dyrkon

**Comment by stejskalleos** [comment]
Katello failures are not related

**Pull request merged**